### PR TITLE
Bump minimum required CMake version to 3.9.

### PR DESCRIPTION
**Pull request by nevack**
Remove old CMake version checks.

### CODE REVIEW

**Comment by nevack** [comment]
Dammit, Solaris/amd64 uses 3.9.1

**Comment by ckerr** [comment]
The newer features are definitely nice-to-have, but CMake 3.14.0 is only a couple of years old. IMO that's too new for this

**Comment by nevack** [comment]
> The newer features are definitely nice-to-have, but CMake 3.14.0 is only a couple of years old. IMO that's too new for this

Changed it to 3.9, as Solaris build host has 3.9.1.

**Comment by ckerr** [comment]
@mikedld opinions?

**Comment by RobFog** [comment]
Typo in commit message and PR title: "minimim"

**Review by mikedld — COMMENTED** [review]
Apart from one comment, LGTM. When I looked at it a while back though, it seemed okay for me to bump to 3.12 (based on versions reported by DistroWatch for various distros). Certainly, there're some cool improvements in 3.14 I'd like to have myself, but... not sure.

I have already disabled GTK client build on Solaris due to outdated dependencies; will look into upgrading the packages (but there was nothing new as of about a month ago), manually installing a more recent CMake version, or dropping the system (until maybe 11.5 or 12 gets released, which I'm not discarding since Oracle will support 11.x until November 2034 [[1]](https://en.wikipedia.org/wiki/Oracle_Solaris#Version_history)), while still leaving OpenIndiana.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
-project(transmission)
-
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
 
-set(CMAKE_MACOSX_RPATH ON)
+# See: https://cmake.org/cmake/help/latest/policy/CMP0048.html
+if (POLICY CMP0048)
+  cmake_policy(SET CMP0048 NEW)
+endif (POLICY CMP0048)
```
[the remark this answers is not in the document]
Comment (nevack) [reply CMakeLists.txt]: Fixed

**Comment by nevack** [comment]
> Typo in commit message and PR title: "minimim"

Fixed

**Review by mikedld — APPROVED** [review]

**Pull request merged**